### PR TITLE
Convert to validate pkg

### PR DESCRIPTION
**Pull request by thockin**
Start chipping away at the tangled web that is API validation.  This introduces a new "validate" package which can become the home for any non-API-type specific validator functions, all with the same naming and calling conventions.  This leaves callers saying things like:

`errs = append(errs, validate.DNSLabel(obj.Field, fldPath.Child("field"))...)`

The goal being to provide util validators for most common validations, even things like numeric comparisons, so we have consistent error messages and types.

xref #56230

```release-note
NONE
```

### CODE REVIEW

**Comment by thockin** [comment]
For history's sake: everything up to InRange() is now green :)  I may add more commits.

**Comment by php-coder** [comment]
Consider converting https://github.com/kubernetes/kubernetes/blob/f0e337cd56f36a6c9b8a0107084baa18fc20ddd8/pkg/apis/extensions/validation/validation.go#L877-L882 to use `validate.NonNegative()`

File: staging/src/k8s.io/apimachinery/pkg/api/validate/validate.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+// NonNegativeDuration validates that given Duration is not negative.
+func NonNegativeDuration(value time.Duration, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if int64(value) < 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath, value, `must be greater than or equal to 0`))
+	}
+	return allErrs
+}
```
Comment (php-coder) [inline staging/src/k8s.io/apimachinery/pkg/api/validate/validate.go]: Consider replacing `NonNegativeDuration` body by invocation of `NonNegative(int64(value), fldPath)`.

Or at least, I'd replace `must be greater than or equal to 0` by `isNegativeErrorMsg` constant.

**Comment by errordeveloper** [comment]
@thockin have you tried rebasing this yet?

**Pull request closed without merging**